### PR TITLE
Self-generate random strings without going through a UUID.

### PR DESCRIPTION
**Pull request by markusthoemmes**
## Description

ActivationIDs are generated via UUID.randomUUID() today. Looking through the code of that you'll notice that it only generates a random hexadecimal String, delimited with dashes. We then proceed to cut out the dashes to end up at our ActivationId format.

We can straight up jump to generate the String ourselves. This even has the added benefit of adding more entropy to that String, because the UUID contains some bytes to determine its version (which are fixed bytes).

## Related issue and scope

- [x] I opened an issue to propose and discuss this change (#3395)

## My changes affect the following components

- [ ] API
- [X] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [X] "Bug fix" (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [X] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [X] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [X] ~~I added tests to cover my changes.~~ (Not necessary as caught by existing tests)
- [X] ~~My changes require further changes to the documentation.~~
- [X] I updated the documentation where necessary.

### CODE REVIEW

**Comment by markusthoemmes** [comment]
Based on a discussion with @rabbah I benchmarked the code (and vastly improved it). Here are some numbers:

```scala
object Bench {
  def test(name: String, count: Int, generator: => String) = {
    println(s"running $name")
    val start = System.currentTimeMillis()
    Seq.fill(count)(generator)
    println(s"took ${System.currentTimeMillis() - start}")
  }

  def main(args: Array[String]): Unit = {
    val secure = new scala.util.Random(new SecureRandom())

    println("warmup")

    test("uuids", 10000, UUIDs.randomUUID().toString.filterNot(_ == '-'))
    test("randomHex", 10000, RandomString.generateHexadecimal(32))

    test("uuids", 10000000, UUIDs.randomUUID().toString.filterNot(_ == '-'))
    test("randomHex", 10000000, RandomString.generateHexadecimal(32))
  }
}
```

```
warmup
running uuids
took 142
running randomHex
took 181
running uuids
took 41861
running randomHex
took 31486
```

You can see that the `randomHex` code is indeed faster now. This is more about that code not allocating as much garbage though.

@rabbah WDYT? I'm a bit unsure if this is even worth it 😁.

**Comment by rabbah** [comment]
bigger 🐟 in the noise...

File: common/scala/src/main/scala/whisk/common/RandomString.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ */
+
+package whisk.common
+
+import java.security.SecureRandom
+
+import org.apache.commons.codec.binary.Hex
+
+object RandomString {
+  private val generator = new ThreadLocal[SecureRandom] {
```
Comment (chetanmeh) [inline common/scala/src/main/scala/whisk/common/RandomString.scala]: May be use [ThreadLocalRandom][1]. As we set `java.security.egd=file:/dev/./urandom` using normal Random may be just fine

[1]:  https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadLocalRandom.html

Comment (markusthoemmes) [reply common/scala/src/main/scala/whisk/common/RandomString.scala]: AFAIK `ThreadLocalRandom` uses a pseudo-random generator vs. `SecureRandom` actually uses `java.security.egd=file:/dev/./urandom`

Comment (chetanmeh) [reply common/scala/src/main/scala/whisk/common/RandomString.scala]: Checked again ... ThreadLocalRandom uses pseudo-random. So makes sense to implement a custom ThreadLocal impl

**Comment by rabbah** [comment]
Leaving as UUID factory based for now. Reopen if you disagree.

**Pull request closed without merging**